sherpa/image/DS9.py

```diff
@@ -333,12 +333,12 @@ def xpaset(cmd, data=None, dataFunc=None, template=_DefTemplate,
         stderr=subprocess.STDOUT,
     )
     try:
-        if dataFunc:
-            dataFunc(p.stdin)
-        elif data:
+        if isinstance(data, six.string_types):
```

Comment:
In order to fix the Python 3 incompatibilities, which are related to the use of `numpy.ndarray.tofile`, I had to create the data with `tobytes` on line 553: f3b3a11b4553779cdc379a16f29bf5e4R553.

In principle, we might do without `dataFunc`, since users can pass the data already as a string or byte array, as done here. However, if there are any users using this argument (I seriously doubt that) chances are their code is broken in Python 3 just as ours was.

However, I'll wait for a round of review: we could simply deprecate the `dataFunc` argument so we don't have to change the API. We probably need to slightly update the doc string, though.